app/services/city_code_generator.rb

```diff
@@ -0,0 +1,8 @@
+# putting this in a separate class because
+# I expect that we can make this logic smarter
+# and use the name of the city to generate a code
+class CityCodeGenerator
+  def self.generate
+    (0...8).map { ('a'..'z').to_a[rand(26)] }.join
```

Comment:
We should probably use [SecureRandom](http://ruby-doc.org/stdlib-1.9.3/libdoc/securerandom/rdoc/SecureRandom.html). It's in stdlib since 1.8? I think. `SecureRandom.hex` should work. If we get a collision so help me god.